client.go

```diff
@@ -118,8 +118,8 @@ func NewClient(pk cipher.PubKey, sk cipher.SecKey, dc disc.APIClient, conf *Conf
 	}
 
 	// Init callback: on delete session.
-	c.EntityCommon.delSessionCallback = func(ctx context.Context, sessionCount int) error {
-		err := c.EntityCommon.updateClientEntry(ctx, c.done)
+	c.EntityCommon.delSessionCallback = func(ctx context.Context) error {
```

Comment:
This callback seems to be incorretly named given that we do not delete a session but remove the whole entry. Also, it appears to me that logically this is incorrect. If a session gets deleted that does not mean the whole `dmsg.Client` should be deregistered. A client can have multiple sessions and if one of them goes down, we should not remove the whole entry but just update as we did previously. You may check the `updateClientEntry` func for reference what we do here. The `updateClientEntry` func also seems to include some logic we can remove which relates to the heartbeat.

Reply:
Done